snuba/utils/streams/batching.py

```diff
@@ -330,29 +294,28 @@ def _commit(self) -> None:
                     raise
 
         if self.commit_log_topic:
-            for item in offsets:
-                if item.offset in self.LOGICAL_OFFSETS:
+            for stream, offset in offsets.items():
+                # XXX: This is an abstraction leak from the Kafka consumer.
+                if offset in self.LOGICAL_OFFSETS:
                     logger.debug(
-                        "Skipped publishing logical offset (%r) to commit log for %s/%s",
-                        item.offset,
-                        item.topic,
-                        item.partition,
+                        "Skipped publishing logical offset (%r) to commit log for %s",
+                        offset,
+                        stream,
                     )
                     continue
-                elif item.offset < 0:
+                elif offset < 0:
                     logger.warning(
-                        "Found unexpected negative offset (%r) after commit for %s/%s",
-                        item.offset,
-                        item.topic,
-                        item.partition,
+                        "Found unexpected negative offset (%r) after commit for %s",
+                        offset,
+                        stream,
                     )
 
                 assert self.producer is not None  # XXX: Hack to ensure non-Optional
                 self.producer.produce(
                     self.commit_log_topic,
                     key="{}:{}:{}".format(
-                        item.topic, item.partition, self.group_id
+                        stream.topic, stream.partition, self.group_id
                     ).encode("utf-8"),
-                    value="{}".format(item.offset).encode("utf-8"),
+                    value="{}".format(offset).encode("utf-8"),
                     on_delivery=self._commit_message_delivery_callback,
                 )
```

Comment:
All of this is going to be moved into a `Consumer` subclass in a follow up, so all of this leaky abstraction will eventually be encapsulated in `snuba.utils.streams.kafka` instead of here.